plugins/BEdita/Core/src/Shell/StreamsShell.php

```diff
@@ -77,4 +93,83 @@ public function removeOrphans()
         }
         $this->out(sprintf('%d stream(s) deleted', $count));
     }
+
+    /**
+     * Re-read streams metadata from file and update information in database.
+     *
+     * @return void
+     */
+    public function refreshMetadata()
+    {
+        $conditions = [
+            'OR' => [
+                $this->Streams->aliasField('file_size') . ' IS' => null,
+                $this->Streams->aliasField('width') . ' IS' => null,
+                $this->Streams->aliasField('height') . ' IS' => null,
+            ],
+        ];
+        $force = (bool)$this->param('force');
+        if ($force) {
+            $conditions = [];
+        }
+
+        $count = $this->Streams->find('all', compact('conditions'))->count();
```

Comment:
```suggestion
        $query = $this->Streams->find('all');
        if ((bool)$this->param('force') === false) {
            $query = $query->where(function (QueryExpression $exp): QueryExpression {
                return $exp->_or(function (QueryExpression $exp): QueryExpression {
                    return $exp
                        ->isNull($this->Streams->aliasField('file_size'))
                        ->isNull($this->Streams->aliasField('width'))
                        ->isNull($this->Streams->aliasField('height'));
                });
            });
        }

        $count = $query->count();
```

Then, the same `$query` object can be safely passed to the `streamsGenerator` method, which can clone it and use to iterate over all streams.